pkgs/cupertino_http/lib/src/cupertino_api.dart

```diff
@@ -664,6 +721,108 @@ class URLSessionDownloadTask extends URLSessionTask {
   String toString() => _toStringHelper('URLSessionDownloadTask');
 }
 
+/// A task associated with a WebSocket connection.
+///
+/// See [NSURLSessionWebSocketTask](https://developer.apple.com/documentation/foundation/nsurlsessionwebsockettask)
+class URLSessionWebSocketTask extends URLSessionTask {
+  final ncb.NSURLSessionWebSocketTask _urlSessionWebSocketTask;
+
+  URLSessionWebSocketTask._(ncb.NSURLSessionWebSocketTask super.c)
+      : _urlSessionWebSocketTask = c,
+        super._();
+
+  /// The close code set when the WebSocket connection is closed.
+  ///
+  /// See [NSURLSessionWebSocketTask.closeCode](https://developer.apple.com/documentation/foundation/nsurlsessionwebsockettask/3181201-closecode)
+  int get closeCode => _urlSessionWebSocketTask.closeCode;
+
+  /// The close reason set when the WebSocket connection is closed.
+  /// If there is no close reason available this property will be null.
+  ///
+  /// See [NSURLSessionWebSocketTask.closeReason](https://developer.apple.com/documentation/foundation/nsurlsessionwebsockettask/3181202-closereason)
+  Data? get closeReason {
+    final reason = _urlSessionWebSocketTask.closeReason;
+    if (reason == null) {
+      return null;
+    } else {
+      return Data._(reason);
+    }
+  }
+
+  /// Sends a single WebSocket message.
+  ///
+  /// Throws an [Error] on failure.
```

Comment:
Nit: Strictly speaking, it completes the Future with an error. `completeError` only becomes a throw when the user awaits the Future. If they use the `then` method instead, they can add an `onError` handler.

Reply:
Done!